### ISSUE TITLE
catkin_download_test_data doesn't update files when MD5 sum changes

### ISSUE THREAD

**Issue by trainman419** [ros/catkin#690]
If the MD5 sum on a file changes or the file is corrupted locally, catkin_download_test_data does not run the download_checkmd5.py script to check it and download a clean copy.

To reproduce (in a workspace with ros_comm):

```
catkin_make tests
md5sum ./build/ros_comm/test/test_rosbag/test/test_rosbag_latched_pub.bag
echo 1 > ./build/ros_comm/test/test_rosbag/test/test_rosbag_latched_pub.bag
catkin_make tests
md5sum ./build/ros_comm/test/test_rosbag/test/test_rosbag_latched_pub.bag
```

**Comment by dirk-thomas** [comment]
Since the download and checksum check are only run once to fetch the file it will be difficult to repeatly check if the file has been corrupted afterwards. Please feel free to provide a pull request for this feature - I might not have the time to implement anything like this in the foreseeable future.

**Comment by dirk-thomas** [comment]
Addressed with #716 being merged.
